File: MY_REPOS/web-dev-notes-resource-site/2-content/Projects/guessing-game-project-master/guessing-game-project-master/terminal-version/python/guessing-game.py
from random import randint

min_num = int(input("Pick a minimum number! "))
max_num = int(input("Pick a maximum number! "))
secret_number = randint(min_num, max_num)

num_attempts = int(input("How many turns do you want? "))

while num_attempts != -1:
    user_guess = int(
        input(f"Try to guess the secret number, you have {num_attempts} left!  ")
    )

    if user_guess < min_num or user_guess > max_num:
        print("That number is not in range")
    elif user_guess < secret_number:
        print(f"Your guess of {user_guess} is too low!")
        num_attempts -= 1
    elif user_guess > secret_number:
        print(f"Your guess of {user_guess} is too high!")
        num_attempts -= 1
    else:
        print("You guessed correct, you win!")
        break

    if num_attempts == 0:
        print("You are out of turns... Better luck next time!")
        break

File: MY_REPOS/Lambda-Resource-Static-Assets/1-projects/my-projects/guessing-game-project-master/guessing-game-project-master/terminal-version/javascript/guessing-game.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let secretNumber;
let numAttempts;

const randomInRange = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min) + min);
};

const checkGuess = (num) => {
  if (num > secretNumber) {
    console.log("Too high");
    return false;
  } else if (num < secretNumber) {
    console.log("Too low");
    return false;
  }
  console.log("Correct!");
  return true;
};

const askGuess = () =>
  rl.question("Enter a Guess, you have  " + numAttempts + " left ", (input) => {
    if (isNaN(Number(input))) {
      console.log("That's not a number!");
      return askGuess();
    }
    let answer = checkGuess(Number(input));
    numAttempts--;
    if (!answer && numAttempts > 0) {
      askGuess();
    } else {
      if (!answer && numAttempts === 0) {
        rl.close();
        console.log("No more turns! You lose!");
      } else {
        rl.close();
        console.log("You Win!");
      }
    }
  });

const askRange = () =>
  rl.question("Enter a max number ", (input) => {
    if (isNaN(Number(input))) {
      console.log("That's not a number!");
      askRange();
    }
    let maxNumber = input;
    console.log("*" + maxNumber + "*");
    rl.question("Enter a min number ", (input) => {
      if (isNaN(Number(input))) {
        console.log("That's not a number!");
        askRange();
      }
      let minNumber = input;
      console.log("*" + minNumber + "*");
      secretNumber = randomInRange(Number(minNumber), Number(maxNumber));
      askGuess();
    });
  });

const askTurnLimit = () =>
  rl.question("How many turns do you want? ", (input) => {
    if (isNaN(Number(input))) {
      console.log("That's not a number!");
      askTurnLimit();
    }
    numAttempts = input;
    askRange();
  });

askTurnLimit();
